sound/soc/sof/topology.c

```diff
@@ -2286,32 +2227,6 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	process->size = ipc_data_size;
 	swidget->private = process;
 
-	ret = sof_ipc_tx_message(sdev->ipc, process->comp.hdr.cmd, process,
-				 ipc_size, r, sizeof(*r));
-
-	if (ret < 0) {
-		dev_err(scomp->dev, "error: create process failed\n");
-		goto err;
-	}
-
-	/* we sent the data in single message so return */
-	if (ipc_data_size)
-		goto out;
```

Comment:
Hmm, while trying to understand readback_offset, I noticed this is removing some special logic in case ipc_data_size is larger than IPC message. In the code you remove, if there's more data, ipc_data_size is set to zero in above code L2245 and we send an IPC message out. 

Here it seems we store the data only if it fits a single IPC message. This seems like the only place where we call sof_get_control_data(), so it seems this init control data is lost.

Reply:
ahh good catch. let me fix this.

Reply:
fixed now. process->size should be set to ipc_size always

Reply:
@kv2019i sorry I spoke too soon. The original setting was correct. process->size = ipc_data_size when the total size is less than IPC_MAX. otherwise it should be 0

Reply:
@ranj063 Hmm, but where is this init data passed now if it doesn't with the IPC msg?

Reply:
If the control data size is larger than max ipc size, it will be sent when widget controls are set up when the pcm is started. If not, it's copied as part of the process_upc_comp and the ipc_data_dize is set here accordingly